pkg/core/template/asset_gear_loader.go

```diff
@@ -0,0 +1,65 @@
+package template
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/skygeario/skygear-server/pkg/core/errors"
+)
+
+type AssetGearLoader struct {
+	AssetGearEndpoint  string
+	AssetGearMasterKey string
+}
+
+func (l *AssetGearLoader) Load(u *url.URL) (templateContent string, err error) {
+	// The url is asset-gear:///assetname
+	path := u.Path
+	assetName := strings.TrimPrefix(path, "/")
+	reqBody := map[string]interface{}{
```

Comment:
Can we define a local struct instead for request & response? It is rather hard to read.